src/main/java/net/mcreator/ui/dialogs/AddBlockPropertyDialog.java

```diff
@@ -39,19 +39,20 @@
 import java.util.List;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Supplier;
+import java.util.stream.Stream;
 
 public class AddBlockPropertyDialog {
 
 	@Nullable
-	public static PropertyDataWithValue<?> showDialog(MCreator mcreator, List<PropertyData<?>> currentEntries,
+	public static PropertyDataWithValue<?> showDialog(MCreator mcreator, Supplier<Stream<String>> currentEntries,

```

Comment:
More clean solution would be to still accept List<PropertyData<?>> since this is what there is stored at the end of the day and do registry name extraction here